MIGRATING_TO_V8.0.0.MD

```diff
@@ -0,0 +1,133 @@
+# Migrating to V8.0.0
+
+## Double encoding RQL values 
+From version 8.0.0 RQL values are now [double encoded](https://docs.extrahorizon.com/extrahorizon/additional-resources/resource-query-language-rql#double-encoding-of-special-characters) by default when using the rql builder, to support the use of special characters in rql operations.
+
+In your existing application it is likely that you have encoded values provided to the rql builder using methods such as `encodeURIComponent()`. To support the automatic double encoding of special characters, where applicable, these instances of encoding should be removed.
+- Before 8.0.0: `rqlBuilder().eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+- After 8.0.0: `rqlBuilder().eq('diagnosis', 'Hypertension - STAGE 1')`
+
+Any use of an rql builder that does not need to support special characters may remain as it is.
+
+### Reverting double encoding for a single rql builder
+You may disable double encoding for an rql builder by setting the option `rqlBuilder({ doubleEncodeValues: false })`
+- Before 8.0.0: `rqlBuilder().eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+- After 8.0.0: `rqlBuilder({ doubleEncodeValues: false }).eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+
+### Reverting double encoding for all rql builders
+The automatic double encoding of values can be reverted to behavior before 8.0.0 using the snippet `rqlBuilder.doubleEncodeValues = false`. This should normally exist alongside the declaration of your Extra Horizon SDK instance at the root of your application.
+- You may then enable double encoding for an rql builder `rqlBuilder({ doubleEncodeValues: true }).eq('diagnosis', 'Hypertension - STAGE 1')`
+- You may manually encode values for an rql builder `rqlBuilder().eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+
+For more information regarding manual double encoding please refer to the [RQL documentation](https://docs.extrahorizon.com/extrahorizon/additional-resources/resource-query-language-rql#double-encoding-of-special-characters).
+
+## Disabling custom key normalization
+
```

Comment:
(minor) I think it could be helpful to have a higher level introduction here. Something mentioning:
- Some endpoints accept or return data not defined by us
- Previously this custom data was treated the same way as data defined by us and still automatically transformed
- This could have led to inconsistencies between what was given to the SDK and returned by the SDK

Reply:
We might be able to reuse stuff from the FibriCheck message 🤔 Always be lazy


> Starting from version 8.0.0 custom properties won’t be affected anymore by the automatic normalization mechanisms. For example everything in the `customFields` property of the profile service will be the same as in the request to and the response of the service. Other examples include the fields of general configuration like `userConfiguration` or `groupConfiguration` since the properties there are also not defined by Extra Horizon. We are currently identifying a list of all those properties. This can also be turned on again on client level `createClient({ ...options, normalizeCustomData: true })` or turned on for a specific request with `await sdk.profiles.find({ rql, normalizeCustomData: true });`
> 
> These normalization methods were previously in effect:
> - Converting keys from camel case to snake case and back, as mentioned in the following [thread](https://extrahorizon.slack.com/archives/C04ACCAKP5F/p1695029676919419). For example everything in the customFields property of the profile service will be the same in the request to and the response from the service.
> - Converting the custom properties with a certain set of timestamp names to a date object. Providing a date to a custom property will now be stored an returned as a string.
> - Converting records_affected & recordsAffected in custom properties to affectedRecords.  (probably never happened, but was a possibility)